design_template.md

```diff
@@ -0,0 +1,536 @@
+SHOP MANAGER Model and Repository Classes Design Recipe
```

Comment:
Well thought out names of classes, variables and methods that are consistent throughout the code.